src/common/config/config.service.ts

```diff
@@ -12,6 +13,15 @@ export class ConfigService extends ConfigServiceSource<EnvironmentVariables> {
   }
 
   public get<T extends keyof EnvironmentVariables>(key: T): EnvironmentVariables[T] {
-    return super.get(key, { infer: true }) as EnvironmentVariables[T];
+    const value = super.get(key, { infer: true }) as EnvironmentVariables[T]
```

Comment:
Maybe it is better not to change this function and read password directly in places there we need postgres password as app.module.

Reply:
And make DB_PASSWORD_FILE required if DB_PASSWORD wasnt provided

Reply:
or i can suggest this:

 ```typescript 
  public get<T extends keyof EnvironmentVariables>(key: T): EnvironmentVariables[T] {
    const value = super.get(key, { infer: true }) as EnvironmentVariables[T];
    if (key == 'DB_PASSWORD') {
      if (value) {
        return value;
      }

      const filePath = super.get('DB_PASSWORD_FILE', { infer: true });

      if (!filePath) {
        // add logger here instead of console
        console.error(`DB_PASSWORD or 'DB_PASSWORD_FILE environments are not provided.`);
        process.exit(1);
      }
      return readFileSync(filePath, 'utf-8') as EnvironmentVariables[T];
    }

    if (key == 'DB_PASSWORD_FILE') {
      const filePath = super.get('DB_PASSWORD_FILE', { infer: true }) as EnvironmentVariables[T];

      return filePath;
    }

    return value;
  }
  ```

Reply:
Thanks for the suggestion, fixed